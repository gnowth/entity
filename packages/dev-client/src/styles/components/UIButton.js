import { color } from '@gnowth/style';
import { css } from 'styled-components';

export const contained = css`
  color: blue;
`;

export const outlined = css`
  border: 1px solid currentColor;
  border-radius: 3px;
  color: ${color({ name: 'primary' })};
  cursor: pointer;
  padding: 5px 30px;
`;

export const text = css`
  color: ${color({ name: 'black' })};
  text-decoration: none;
`;

export const toggle = css`
  color: blue;
`;
