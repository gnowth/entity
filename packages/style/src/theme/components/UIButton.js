import { css } from 'styled-components';

import { color, colorFromPalette, boxshadow } from '../../selectors';

export const component_uiButton_contained = {
  contentComponentPaletteAsBackground: true,
  iconComponentPaletteAsBackground: true,
  processingComponentPaletteAsBackground: true,
  css: css`
    background-color: ${colorFromPalette()};

    &:hover {
      background-color: ${colorFromPalette({ weight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ weight: '700' })};
    }

    &:disabled {
      background-color: ${color({ name: 'gray', weight: '200' })};
      cursor: auto;

      * {
        color: ${color({ name: 'gray', weight: '800' })};
      }
    }
  `,
};

export const component_uiButton_fab = {
  contentComponentHidden: true,
  contentComponentPaletteAsBackground: true,
  iconComponentHidden: false,
  iconComponentPaletteAsBackground: true,
  processingComponentPaletteAsBackground: true,
  css: css`
    ${boxshadow({ name: 'materialHover3' })};
    background-color: ${colorFromPalette()};
    border-radius: 50%;
    padding: 0;
    height: 3em;
    width: 3em;

    &:hover {
      background-color: ${colorFromPalette({ weight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ weight: '700' })};
    }

    &:disabled {
      background-color: ${color({ name: 'gray', weight: '200' })};
      cursor: auto;

      * {
        color: ${color({ name: 'gray', weight: '800' })};
      }
    }
  `,
};

export const component_uiButton_icon = {
  contentComponentHidden: true,
  iconComponentHidden: false,
  css: css`
    border-radius: 50%;
    padding: 0.5em;

    ${props => props.media && css`
      @media screen and ${props.media} {
        padding: 0.5em;
      }
    `}
  `,
};

export const component_uiButton_outlined = {
  css: css`
    border: 1px solid ${color({ name: 'gray', weight: '200' })};

    &:hover {
      background-color: ${color({ name: 'gray', weight: '50' })};
    }

    &:disabled {
      background-color: ${color({ name: 'white' })};
      border-color: ${color({ name: 'gray', weight: '800' })};
      cursor: auto;

      * {
        color: ${color({ name: 'gray', weight: '400' })};
      }
    }
  `,
};

export const component_uiButton_raised = {
  contentComponentPaletteAsBackground: true,
  iconComponentPaletteAsBackground: true,
  processingComponentPaletteAsBackground: true,
  css: css`
    ${boxshadow({ name: 'material2' })};
    background-color: ${colorFromPalette()};

    &:hover {
      background-color: ${colorFromPalette({ weight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ weight: '700' })};
    }

    &:disabled {
      background-color: ${color({ name: 'gray', weight: '200' })};
      cursor: auto;

      * {
        color: ${color({ name: 'gray', weight: '800' })};
      }
    }
  `,
};
