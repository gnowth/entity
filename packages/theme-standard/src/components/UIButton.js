import { color, colorFromPalette, boxshadow } from '@gnowth/theme';
import { css } from 'styled-components';

export const component_uiButton_contained = {
  contentPaletteAsBackground: true,
  iconPaletteAsBackground: true,
  processingPaletteAsBackground: true,
  css: css`
    background-color: ${colorFromPalette()};
    min-width: 8rem;

    &:hover {
      background-color: ${props => colorFromPalette({ paletteWeight: '400' })(props)};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_fab = {
  contentHidden: true,
  contentPaletteAsBackground: true,
  iconHidden: false,
  iconPaletteAsBackground: true,
  processingPaletteAsBackground: true,
  css: css`
    ${boxshadow({ name: 'materialHover3' })};
    background-color: ${colorFromPalette()};
    border-radius: 50%;
    padding: 0;
    height: 3em;
    width: 3em;

    &:hover {
      background-color: ${colorFromPalette({ paletteWeight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_flat = {
  contentPaletteAsBackground: true,
  iconPaletteAsBackground: true,
  processingPaletteAsBackground: true,
  css: css`
    background-color: ${colorFromPalette()};
    border-radius: 0;

    &:hover {
      background-color: ${colorFromPalette({ paletteWeight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_icon = {
  contentHidden: true,
  iconHidden: false,
  css: css`
    border-radius: 50%;
    padding: 0.5em;

    ${props => props.media && css`
      @media screen and ${props.media} {
        padding: 0.5em;
      }
    `}
  `,
};

export const component_uiButton_navigation = {
  css: css`
    border-radius: 0;

    &.active {
      border-bottom: 2px solid ${colorFromPalette()};
    }
  `,
};

export const component_uiButton_outlined = {
  css: css`
    border: 1px solid ${color({ palette: 'gray', paletteWeight: '200' })};

    &:hover {
      background-color: ${color({ palette: 'gray', paletteWeight: '50' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'white' })};
      border-color: ${color({ palette: 'gray', paletteWeight: '800' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '400' })};
      }
    }
  `,
};

export const component_uiButton_raised = {
  contentPaletteAsBackground: true,
  iconPaletteAsBackground: true,
  processingPaletteAsBackground: true,
  css: css`
    ${boxshadow({ name: 'material2' })};
    background-color: ${colorFromPalette()};

    &:hover {
      background-color: ${colorFromPalette({ paletteWeight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_text = {};
