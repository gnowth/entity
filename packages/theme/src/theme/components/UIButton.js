import { css } from 'styled-components';

import { color, colorFromPalette, boxshadow } from '../../selectors';

export const component_uiButton_contained = {
  contentComponentPaletteAsBackground: true,
  iconComponentPaletteAsBackground: true,
  processingComponentPaletteAsBackground: true,
  css: css`
    background-color: ${colorFromPalette()};

    &:hover {
      background-color: ${props => colorFromPalette({ paletteWeight: '400' })(props)};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_fab = {
  contentComponentHidden: true,
  contentComponentPaletteAsBackground: true,
  iconComponentHidden: false,
  iconComponentPaletteAsBackground: true,
  processingComponentPaletteAsBackground: true,
  css: css`
    ${boxshadow({ name: 'materialHover3' })};
    background-color: ${colorFromPalette()};
    border-radius: 50%;
    padding: 0;
    height: 3em;
    width: 3em;

    &:hover {
      background-color: ${colorFromPalette({ paletteWeight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_flat = {
  contentComponentPaletteAsBackground: true,
  iconComponentPaletteAsBackground: true,
  processingComponentPaletteAsBackground: true,
  css: css`
    background-color: ${colorFromPalette()};
    border-radius: 0;

    &:hover {
      background-color: ${colorFromPalette({ paletteWeight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_icon = {
  contentComponentHidden: true,
  iconComponentHidden: false,
  css: css`
    border-radius: 50%;
    padding: 0.5em;

    ${props => props.media && css`
      @media screen and ${props.media} {
        padding: 0.5em;
      }
    `}
  `,
};

export const component_uiButton_navigation = {
  css: css`
    border-radius: 0;

    &.active {
      border-bottom: 2px solid ${colorFromPalette()};
    }
  `,
};

export const component_uiButton_outlined = {
  css: css`
    border: 1px solid ${color({ palette: 'gray', paletteWeight: '200' })};

    &:hover {
      background-color: ${color({ palette: 'gray', paletteWeight: '50' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'white' })};
      border-color: ${color({ palette: 'gray', paletteWeight: '800' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '400' })};
      }
    }
  `,
};

export const component_uiButton_raised = {
  contentComponentPaletteAsBackground: true,
  iconComponentPaletteAsBackground: true,
  processingComponentPaletteAsBackground: true,
  css: css`
    ${boxshadow({ name: 'material2' })};
    background-color: ${colorFromPalette()};

    &:hover {
      background-color: ${colorFromPalette({ paletteWeight: '400' })};
    }

    &:focus {
      background-color: ${colorFromPalette({ paletteWeight: '700' })};
    }

    &:disabled {
      background-color: ${color({ palette: 'gray', paletteWeight: '200' })};
      cursor: auto;

      * {
        color: ${color({ palette: 'gray', paletteWeight: '800' })};
      }
    }
  `,
};

export const component_uiButton_text = {};
