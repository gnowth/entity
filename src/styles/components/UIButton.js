import { css } from 'styled-components';

export const contained = css`
  color: blue;
`;

export const outlined = css`
  border: 1px solid currentColor;
  border-radius: 3px;
  color: ${props => props.theme.vars.colorPrimary};
  cursor: pointer;
  padding: 5px 30px;
`;

export const text = css`
  color: ${props => props.theme.vars.colorBlack};
  text-decoration: none;
`;

export const toggle = css`
  color: blue;
`;
