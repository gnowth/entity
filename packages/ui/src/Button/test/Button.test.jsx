import 'jest-styled-components';
import renderer from 'react-test-renderer';
import React from 'react';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';

import UIButton from '../Button';

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={{}}>
      <IntlProvider locale="en" messages={{}}>
        <UIButton content="Test" />
      </IntlProvider>
    </ThemeProvider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
