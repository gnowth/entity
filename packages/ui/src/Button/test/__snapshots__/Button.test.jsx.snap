// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1.c1 {
  font-size: 1rem;
}

.c4 {
  -webkit-animation-name: bvvTNa;
  animation-name: bvvTNa;
  fill: transparent;
  stroke: black;
  stroke-dasharray: 251.32741228718345px;
  stroke-width: 20%;
  -webkit-transition: stroke-dashoffset 225ms linear;
  transition: stroke-dashoffset 225ms linear;
  -webkit-transform-box: view-box;
  -ms-transform-box: view-box;
  transform-box: view-box;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation-duration: 4s;
  animation-duration: 4s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: cubic-bezier(0.35,0,0.25,1);
  animation-timing-function: cubic-bezier(0.35,0,0.25,1);
  -webkit-transition-property: stroke;
  transition-property: stroke;
}

.c3 {
  height: 100%;
  left: 0;
  top: 0;
  overflow: visible;
  position: absolute;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  width: 100%;
}

.c2 {
  display: inline-block;
  height: 32px;
  position: relative;
  width: 32px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 0;
  border-radius: 0.25em;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  padding: 0.5em 1.5em;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  background-color: black;
}

.c0:focus {
  background-color: black;
}

.c0:disabled {
  background-color: black;
  cursor: auto;
}

.c0:disabled * {
  color: black;
}

@media print {
  .c5 {
    color: black;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<button
  className="c0"
  content="Test"
>
  <i
    className="uiButton-Icon icon icon-name uiButton-Icon c1"
    hidden={true}
    name="name"
  />
  <div
    className="c2"
    hidden={true}
    size="32px"
    value={null}
  >
    <svg
      className="c3"
      focusable="false"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 100 100"
    >
      <circle
        className="c4"
        cx="50%"
        cy="50%"
        r="40"
        value={null}
      />
    </svg>
  </div>
  <span
    className="uiButton-Content c5"
  >
    Test
  </span>
</button>
`;
